wallet/src/test/java/io/neow3j/wallet/WalletTest.java

```diff
@@ -93,32 +92,24 @@ public void addAccount() throws InvalidAlgorithmParameterException,
             NoSuchAlgorithmException, NoSuchProviderException {
 
         Wallet w = Wallet.createWallet();
-        Account acct = Account.fromECKeyPair(ECKeyPair.createEcKeyPair()).build();
+        Account acct = new Account(ECKeyPair.createEcKeyPair());
         w.addAccount(acct);
-        assertTrue(w.getAccounts().size() == 2);
+        assertEquals(2, w.getAccounts().size());
         assertEquals(w.getAccount(acct.getScriptHash()), acct);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void failAddingSecondDefaultAccountToWalletWithDefaultAccount()
-            throws InvalidAlgorithmParameterException, NoSuchAlgorithmException,
-            NoSuchProviderException {
-
-        Wallet w = Wallet.createWallet();
-        Account acct = Account.fromECKeyPair(ECKeyPair.createEcKeyPair())
-                .isDefault()
-                .build();
-        w.addAccount(acct);
-    }
-
     @Test
     public void testAddDuplicateAccount() throws InvalidAlgorithmParameterException,
             NoSuchAlgorithmException, NoSuchProviderException {
 
         Wallet w = Wallet.createWallet();
-        Account acct = Account.fromECKeyPair(ECKeyPair.createEcKeyPair()).build();
-        assertTrue(w.addAccount(acct));
-        assertFalse(w.addAccount(acct));
+        Account acct = new Account(ECKeyPair.createEcKeyPair());
+
+        // Add account to wallet.
+        w.addAccount(acct);
+        assertTrue(w.getAccounts().contains(acct));
+        // Adding an account twice does not change the wallet.
+        assertThat(w.addAccount(acct), is(w));
```

Comment:
This assertion is always true because the returned wallet is always the same as the one the method is called on. Only thing we could assert here is if the accounts list still contains only one account.

Reply:
I changed this, to compare the number of accounts in the wallet.